Added real time visualization of shape creation and more improvements

VDykyi opened the pull request:
Added real time visualization of shape creation and more improvements

VDykyi committed:
Added real time visualization of shape creation and more improvements

marcus-sonestedt commented on src/VAC/Global.h:
```diff
@@ -127,6 +127,23 @@ class Q_VPAINT_EXPORT Global: public QObject
     void setEdgeColor(const QColor& newColor);
+    void setFaceColor(const QColor& newColor);
+
+    bool drawShapeFaceEnabled() const;
```
getters for bool props defining a single value should have "is" prefix according to Qt standard.

VDykyi replied:
Done

marcus-sonestedt commented on src/VAC/VectorAnimationComplex/VAC.cpp:
```diff
@@ -5814,25 +5819,67 @@ void VAC::updateCellsToConsiderForCutting()
+
+void VAC::addToHovered(const CellSet &cells, bool emitSignal)
+{
+    for(Cell * c: cells)
```
"auto c" works too

VDykyi replied:
Done

marcus-sonestedt reviewed (approved):
Awesome work there, Vasyl! 👍👍

VDykyi committed:
Fixed sometimes disappearing edges when draw next shape and optimization code

VDykyi commented:
- Implemented the real time visualization when shape creation.
- Added ability to draw "faces"(fill) for closed shapes and ability to adjust it opacity and color.
- Added ability to adjusting of the highlighting and selecting colors and opacity for edges and fill due to current shape color.
- Also realized simplified selection of shapes.
- Added ability to show/hide the rectangle arround the shape during creation.

VDykyi committed:
Added forwarding the geters/setters

VDykyi committed:
Added a lot of improvements
